### ISSUE TITLE
ERROR: read: Input/output error

### ISSUE THREAD

**Issue by borisovg** [adrienverge/openfortivpn#362]
For some reason I started seeing this error yesterday when trying to connect:

```
DEBUG:  Got Address: 10.18.2.1
DEBUG:  if_config: not ready yet...
ERROR:  read: Input/output error
INFO:   Cancelling threads...
DEBUG:  Waiting for pppd to exit...
DEBUG:  waitpid: pppd exit status code 16
INFO:   pppd: The link was terminated by the modem hanging up.
INFO:   Terminated pppd.
INFO:   Closed connection to gateway.
```

Debian Sid, openfortivpn 1.7.1

Updated some packages yesterday so possible something came in and broke stuff.

Looks different to #154 so opened this new issue.

**Comment by DimitriPapadopoulos** [comment]
This message does look strange:
```DEBUG:  if_config: not ready yet...```
The relevant code is:
https://github.com/adrienverge/openfortivpn/blob/a5ab08a/src/io.c#L540-L563
Clearly the PPP interface isn't up.

Now why PPP isn't up? You have provided the PPP log which is extremely helpful. Often users don't provide it and I myself forget about it. We should write a wiki page to document how to troubleshoot _openfortivpn_!
This looks suspect:
```Peer refused to agree to our IP address```
has anything changed on the Fortigate device?

**Comment by borisovg** [comment]
Thanks, I will ask the team that looks after the device.

**Comment by borisovg** [comment]
OK, error went away with me doing nothing - must have been something on the server side. Thanks for your help. :)

**Comment by ageric** [comment]
I experienced the same nonsense when I forgot to use sudo.

Personally, I think openfortivpn should have "Your'e not root. It's frequently needed" as a printout in its path to an error exit (assuming it's true). For me, it would have saved about 40 minutes of post-afterwork-pondering.

**Comment by DimitriPapadopoulos** [comment]
@ageric That looks unrelated. Please open a new ticket.

**Comment by DimitriPapadopoulos** [comment]
@ageric Also _openfortivpn_ prints:
_WARN:   This process was not spawned with root privileges, this will probably not work._
How can we improve on this? We're open to suggestions.

**Comment by DimitriPapadopoulos** [comment]
Opened #373 myself to exit immediately if `geteuid() != 0`.

**Comment by mrbaseman** [comment]
in the recently released version 1.8.0 _openfortivpn_ exits immediately when it is not started with enough privileges (i.e. when it is not started via sudo or direictly launched by root or systemd)
